spec/models/accounting/quickbooks/transaction_creator_spec.rb

```diff
@@ -2,28 +2,75 @@
+  let(:generic_service) { instance_double(Quickbooks::Service::JournalEntry, all: [], create: nil) }
   let(:class_service) { instance_double(Quickbooks::Service::Class, find_by: [class_ref]) }
   let(:connection) { instance_double(Accounting::Quickbooks::Connection) }
-  let(:account) { instance_double(Accounting::Account, qb_id: 98) }
-  subject { described_class.new(instance_double(Division, qb_connection: connection, principal_account: account)) }
+  let(:account) { instance_double(Accounting::Account, qb_id: qb_principal_account_id) }
+  let(:creator) { described_class.new(instance_double(Division, qb_connection: connection, principal_account: account)) }
 
   before do
-    allow(subject).to receive(:service).and_return(generic_service)
-    allow(subject).to receive(:class_service).and_return(class_service)
+    allow(creator).to receive(:service).and_return(generic_service)
+    allow(creator).to receive(:class_service).and_return(class_service)
   end
 
-  it 'should create class ref, and create journal entry' do
-    expect(generic_service).to receive(:create)
-    expect(class_service).to receive(:find_by)
-
-    subject.add_disbursement(
-      amount: 12.09,
-      loan_id: 2,
-      memo: 'I am memo',
-      description: 'I am a line item description',
-      qb_bank_account_id: 89,
-      qb_customer_id: 3
-    )
+  context 'when qb customer does exist' do
+    let(:qb_customer_id) { 3 }
+    let(:loan_id) { 2 }
+    let(:amount) { 78.20 }
+    let(:memo) { 'I am a memo' }
+    let(:description) { 'I am a line item description' }
+    let(:qb_bank_account_id) { 89 }
+    let(:qb_principal_account_id) { 92 }
+    let(:date) { nil }
+
+    subject do
+      creator.add_disbursement(
+        amount: amount,
+        loan_id: loan_id,
+        memo: memo,
+        description: description,
+        qb_bank_account_id: qb_bank_account_id,
+        qb_customer_id: qb_customer_id,
+        date: date
+      )
+    end
+
+    context 'generic_service' do
+      it 'creates JournalEntry with correct data' do
+        expect(generic_service).to receive(:create) do |arg|
+          expect(arg.line_items.count).to eq 2
+          expect(arg.private_note).to eq memo
+          expect(arg.txn_date).to be_nil
+
+          list = arg.line_items
+          expect(list.map(&:amount).uniq).to eq [amount]
+          expect(list.map(&:description).uniq).to eq [description]
+
+          details = list.map { |i| i.journal_entry_line_detail }
+          expect(details.map { |i| i.posting_type }.uniq).to match_array %w(Debit Credit)
+          expect(details.map { |i| i.entity.type }.uniq).to eq %w(Customer)
+          expect(details.map { |i| i.entity.entity_ref.value }.uniq).to eq [qb_customer_id]
+          expect(details.map { |i| i.class_ref.value }.uniq).to eq [loan_id]
+          expect(details.map { |i| i.account_ref.value }.uniq).to match_array [qb_bank_account_id, qb_principal_account_id]
```

Comment:
Quick note: I would use eq here instead of match_array because you know what order they will be in and so you can be sure that the Credit transaction has the qb_bank_account_id, etc.

match_array / contains_exactly are best used when you don't know the order.